File: app/src/main/java/org/pureagave/zodiac/control/core/net/FleetBus.kt
package org.pureagave.zodiac.control.core.net

/**
 * The vehicle's local data bus. One-to-many realtime streams ride **fixed
 * multicast groups** — the group address is a constant baked into the app, so
 * it's independent of whatever the travel router's DHCP hands out (no hardcoded
 * peer IPs, survives dynamic leases). A source announces itself simply by
 * transmitting to its group; every device that joined the group receives it.
 * All link-local (TTL 1) → nothing leaves the vehicle and it works with the
 * internet down.
 *
 * Addresses are in the administratively-scoped 239.0.0.0/8 range (never routed
 * off-link). The beacon (`:beacon` module) mirrors these constants.
 */
object FleetBus {
    /** Vehicle telemetry (NMEA GPS + compass) from the Sensor Hub. */
    const val TELEMETRY_GROUP = "239.7.7.10"
    const val TELEMETRY_PORT = 10110

    /** Thermal detections from the edge box (Jetson). */
    const val THREAT_GROUP = "239.7.7.20"
    const val THREAT_PORT = 10120

    /** Link-local only — one hop, never past the vehicle's own switch/AP. */
    const val TTL = 1
}

File: jetson/zvision/fleet_bus.py
"""Vehicle data-bus constants — the exact mirror of the Kotlin ``FleetBus``
object and the ``:beacon`` module.

Fixed multicast groups in the administratively-scoped 239.0.0.0/8 range (never
routed off-link), TTL 1 so nothing leaves the vehicle. A source announces itself
simply by transmitting to its group; every device that joined receives it — no
hardcoded peer IPs, survives whatever the travel router's DHCP hands out.
"""

# Vehicle telemetry (NMEA GPS + compass) from the Sensor Hub.
TELEMETRY_GROUP = "239.7.7.10"
TELEMETRY_PORT = 10110

# Thermal detections from the edge box (this Jetson).
THREAT_GROUP = "239.7.7.20"
THREAT_PORT = 10120

# Link-local only — one hop, never past the vehicle's own switch/AP.
TTL = 1
